### PR TITLE
UIROLES-75: reduce amount of GET requests by retrieving selected app ids from useRoleCapabilities and useRoleCapabilitySets

### PR DESCRIPTION
**Pull request by aidynoJ**
Refs [UIROLES-75. ](https://folio-org.atlassian.net/browse/UIROLES-75)
Instead of fetching all capabilities and capability sets to get the selected applications, just retrieve them from role capabilities and role capability sets. This improvement was noted during the demo of UIROLES-75.

### CODE REVIEW

**Comment by aidynoJ** [comment]
How it worked previously. retrieve all of the capabilities, capability sets, role capabilities, role capability sets. Iterate over capabilities(capability sets) for each roles capabilities(capability sets), and define selected applications to retrieve data by. 
   I didn't notice before, that we can get data of applicationIds right from the role capabilities, role capability sets. No need to retrieve all of the capabilities, capabilitySets.

File: src/settings/components/CreateEditRole/EditRole.js [hunk 1]
```diff
@@ -104,21 +104,15 @@ const EditRole = ({ path }) => {
     onClose();
   };
 
-  const isInitialDataReady = !isCapabilityListLoading && isRoleCapabilitySetsLoaded
-  && isInitialRoleCapabilitiesLoaded && !isCapabilitySetsLoading;
+  const isInitialDataReady = isRoleCapabilitySetsLoaded && isInitialRoleCapabilitiesLoaded;
```
Comment (zburke) [inline src/settings/components/CreateEditRole/EditRole.js:107]: Petty: Can we name these things the same way if we are checking the same condition? IOW why is `Initial` in one of these names but not the other?

Comment (aidynoJ) [reply src/settings/components/CreateEditRole/EditRole.js]: Fixed, thank you

File: src/hooks/useRoleCapabilitySets.js [hunk 2]
```diff
@@ -30,11 +33,21 @@ const useRoleCapabilitySets = (roleId) => {
...
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [data]);
+
   return { initialRoleCapabilitySetsSelectedMap,
     isSuccess,
     capabilitySetsTotalCount: data?.totalRecords || 0,
     groupedRoleCapabilitySetsByType,
-    capabilitySetsCapabilities };
+    capabilitySetsCapabilities,
+    capabilitySetAppIds };
```
Comment (zburke) [inline src/hooks/useRoleCapabilitySets.js:50]: The parallel value in `useRoleCapabilities` is `capabilitiesAppIds`. Parallel structures should have parallel names. I think `capabilitySetsAppIds` is correct.

Comment (aidynoJ) [reply src/hooks/useRoleCapabilitySets.js]: You are right, fixed

**Review by zburke — COMMENTED** [review]
It is hard to follow exactly what is happening in this refactor, but that is mainly due to not having a great high-level picture of how data-load is happening in the first place. 

IIUC, the data structure of role <<->> capability <<-> application (ditto capability-sets) can be used to derive the list of selected applications, which can further be used to retrieve the list of available capabilities and capability-sets when editing a role. Previously, we just retrieved everything, regardless of whether it was affiliated with a selected application. 

Is that correct?

**Review by zburke — APPROVED** [review]
